Sources/Plasma/FeatureLib/pfPatcher/pfPatcher.cpp

```diff
@@ -476,55 +480,78 @@ void pfPatcherWorker::Run()
 
 void pfPatcherWorker::ProcessFile()
 {
+    plFileName clientDir = plFileSystem::GetCurrentAppPath().StripFileName();
+
     do {
         NetCliFileManifestEntry& entry = fQueuedFiles.front();
+        fQueuedFiles.pop_front();
 
         // eap sucks
-        plFileName clName = ST::string::from_wchar(entry.clientName);
+        plFileName clName = plFileName(ST::string::from_wchar(entry.clientName)).Normalize();
         ST::string dlName = ST::string::from_wchar(entry.downloadName);
 
+        // Only operate on files if they are contained within the client directory...
+        plFileName clAbsPath = clName.AbsolutePath();
+        if (!clAbsPath.AsString().starts_with(clientDir.AsString())) {
```

Comment:
My instinct is telling me that a whitelist approach is better than a blacklist approach.  Something along the lines of "Is the manifest requesting we delete a file we have in our local store?  Then delete the file indicated from the local store."  This way we never use the manifest's supplied name for anything other than the comparison, keeping it at the lowest-possible level of trust.